app/grandchallenge/codebuild/models.py

```diff
@@ -0,0 +1,36 @@
+from django.db import models
+from django.utils.text import get_valid_filename
+
+from grandchallenge.core.models import UUIDModel
+from grandchallenge.core.storage import private_s3_storage
+
+
+def log_path(instance, filename):
+
+    return (
+        f"{instance._meta.app_label.lower()}/"
+        f"{instance._meta.model_name.lower()}/"
+        f"{instance.pk}/"
+        f"{get_valid_filename(filename)}"
+    )
+
+
+class Build(UUIDModel):
+    build_id = models.CharField(max_length=1024)
+    project_name = models.CharField(max_length=1024)
+    status = models.CharField(blank=True, max_length=256)
+    build_log = models.FileField(
+        null=True, upload_to=log_path, storage=private_s3_storage
+    )
```

Comment:
You should be able to read the log files with the `gzip` module and then save them to a TextField.